### PR TITLE
feat(sagemaker): support batch-transform

### PR DESCRIPTION
**Pull request by deepankarm**
## Goals

PR in continuation to https://github.com/jina-ai/jina/pull/6046. 

Sagemaker still calls the `POST /invocations` route during batch-transform but with different content-type & body. To enable both `application/json` (for inference) and `text/csv` (for batch-transform, I removed the pydantic model from the path signature, rather use fastapi's `Request` class and evaluate the body according to the content-type. 

For batch-transform, 

- Expect `text/csv` or `application/csv` content type.
- Validate if the format is correct.
- Convert each row in the csv to `input_model` (user-defined model in the Executor route).
- Invoke the route for all inputs together and return the response. This makes sure the values appended to the output csv by sagemaker has a similar format as inference. 

#### Gotchas

- Sagemaker doesn't support CSV with headers. We follow the same restriction. 
- Rows in the input csv should have all fields defined in the input type (including optional fields like `id`)

  ```python
  from docarray import BaseDoc
  
  class TextDoc(BaseDoc):
      text: str
  ```

  Following inputs are valid

  ```csv
  1,abcd
  2,efgh
  3,ijkl
  4,mnop
  5,qrst
  6,uvwx
  7,yzab
  8,cdef
  9,ghij
  10,klmn
  ```

  Following CSV is invalid.

  ```csv
  abcd
  efgh
  ijkl
  mnop
  qrst
  uvwx
  yzab
  cdef
  ghij
  klmn
  ```

#### End-to-end tests

I've manually tested the Executor after pushing a model to S3, the Executor image to ECR & running batch-transform jobs for both `SingleLine` and `MultiLine` records.

---

- [x] check and update documentation. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#-contributing-documentation) and ask the team.

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]
can u add tests with docker?

**Comment by deepankarm** [comment]
> can u add tests with docker?

Added - https://github.com/jina-ai/jina/pull/6055/commits/a7faa6f412742605ee9af43232c66db9b2175dd5

File: jina/serve/runtimes/worker/http_sagemaker_app.py [hunk 1]
```diff
@@ -115,6 +126,48 @@ async def post(body: input_model, response: Response):
...
+                        status_code=400,
+                        detail='Invalid CSV input. Please check your input.',
+                    )
+
+                # NOTE: Sagemaker only supports csv files without header, so we enforce
+                # the header by getting the field names from the input model.
+                # This will also enforce the order of the fields in the csv file.
+                # This also means, all fields in the input model must be present in the
+                # csv file including the optional ones.
+                field_names = [f for f in input_doc_list_model.__fields__]
```
Comment (JoanFM) [inline jina/serve/runtimes/worker/http_sagemaker_app.py:150]: I believe that `DocArray` provides a `from_csv` method

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/serve/runtimes/worker/http_sagemaker_app.py (hunk 1 above)
Comment (deepankarm) [reply jina/serve/runtimes/worker/http_sagemaker_app.py:150]: Doesn't work without csv header. Sagemaker only supports csvs without header.

File: tests/integration/docarray_v2/sagemaker/test_sagemaker.py [hunk 2]
```diff
@@ -153,6 +169,29 @@ def test_provider_sagemaker_deployment_inference():
             assert len(resp_json['data'][0]['embeddings'][0]) == 64
 
 
+def test_provider_sagemaker_deployment_inference_docker(replica_docker_image_built):
+    with Deployment(uses='docker://sampler-executor', provider='sagemaker', port=12345):
+        # Test the `GET /ping` endpoint (added by jina for sagemaker)
```
Comment (JoanFM) [inline tests/integration/docarray_v2/sagemaker/test_sagemaker.py:174]: please make port random

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/integration/docarray_v2/sagemaker/test_sagemaker.py (hunk 2 above)
Comment (deepankarm) [reply tests/integration/docarray_v2/sagemaker/test_sagemaker.py:173]: Done

File: jina/serve/runtimes/worker/http_sagemaker_app.py (hunk 1 above)
Comment (JoanFM) [reply jina/serve/runtimes/worker/http_sagemaker_app.py:150]: can u open a ticket on DocArray with this?

**Review by JoanFM — APPROVED** [review]

**Review by JoanFM — APPROVED** [review]

**Pull request merged**